blog/2023-03-19-get-started-wtih-celo-using-rust/get-started-with-celo-using-rust.md

```diff
@@ -0,0 +1,249 @@
+---
+title: Get started with CELO using Rust
+description: This article is intended for developers that have some familiarity with Rust and want to construct a Rust program to communicate with the Celo blockchain.
+
+authors:
+  - name: Muhideen Mujeeb
+    title: Software Engineer
+    url: https://github.com/mujhtech
+    image_url: https://avatars.githubusercontent.com/u/41507005?v=4
+tags: [celo,celosage,truffle,smartcontract,solidity, beginner]
+hide_table_of_contents: true
+slug: /tutorials/get-started-with-celo-using-rust
+---
+
+![header](../../src/data-tutorials/showcase/beginner/get-started-with-celo-using-rust.png)
+
+## Introduction
+
+In this tutorial, we'll go over the fundamentals of interacting with Solidity smart contract programming language using web3. You will have a fundamental understanding of how to make a contract call in the rust programming language. 
+
+This tutorial will demonstrate how simple it is to interact with smart contracts, call functions, and listen to events in rust.
+
+### How it works
+
+Web3 Package is a pretty cool crate package that let you deploy, interact with any blockchain network.
+
+## Prerequisites
+
+First, This tutorial assumes that you are already familiar with solidity and understand how smart contracts work and also assumes that you already know the basics of using rust language.
+
+For this project we'll be using a few interesting dependencies:
+
+- [tokio](https://crates.io/crates/tokio)
+- [web3](https://crates.io/crates/web3)
+- [ethers](https://crates.io/crates/ethers)
+
+![Add dependecies](./images/screenshot-1.png)
+
+## Getting Started
+
+I assume anyone going through this tutorial already understands and uses Rust, so I will skip the setup involved in getting Rust to work on your development computer. That means I assume you already have VS Code/Intellij Idea/Eclipse/Atom and Rust setup on your PC.
+
+*If you are entirely new to Rust, here ( [https://www.rust-lang.org/learn/get-started](https://www.rust-lang.org/learn/get-started) ) is a good tutorial you can learn from or make use of online rust playground ( [https://play.rust-lang.org/](https://play.rust-lang.org/) ).
+
+
```

Comment:
There should only be one line spacing before and after each headline. Please remove the extra spacing here and everywhere else in the tutorial. i noticed them in some other places too.